### PR TITLE
Debug streamer: Add downsampling by tsalis

### PR DESCRIPTION
**Pull request by matthias-kleiner**

### CODE REVIEW

File: Detectors/TPC/workflow/src/TPCTimeSeriesSpec.cxx [hunk 1]
```diff
@@ -679,56 +679,73 @@ class TPCTimeSeries : public Task
...
-        auto clusterMask = makeClusterBitMask(track);
-        const auto& trkOrig = tracksTPC[iTrk];
-        const bool isNearestVtx = (idxITSTPC.back() == -1); // is nearest vertex in case no vertex was found
-        const float mx_ITS = hasITSTPC ? tracksITSTPC[idxITSTPC.front()].getX() : -1;
-        int typeSide = 2; // A- and C-Side cluster
-        if (track.hasASideClustersOnly()) {
-          typeSide = 0;
-        } else if (track.hasCSideClustersOnly()) {
-          typeSide = 1;
+        const float sqrts = 13600; // o2::base::GRPGeomHelper::instance().getGRPLHCIF()->getSqrtS();
```
Comment (matthias-kleiner) [inline Detectors/TPC/workflow/src/TPCTimeSeriesSpec.cxx]: Hello @shahor02 
for the downsampling with the tsalis we also need the centre of mass energy, which should be the 13.6TeV for pp.
I tried to get it with `o2::base::GRPGeomHelper::instance().getGRPLHCIF()->getSqrtS();` as you see, but somehow it returned `[2828107:tpc-time-series]: [21:21:45][INFO] sqrts: 113324`. Is there some other way to get the 13.6TeV or am I missing something?

**Comment by miranov25** [comment]
Hello @mkleiner.

I prefer to write motivation here as well, and change the name tilete of the pull request accordingly.

## Code to enable predefined downsampling in the pt and restore the original spectrum using the re-weighting method.
```
//// Tsalis/Hagedorn function describing charged pt spectra (m s = 62.4 GeV to 13 TeV) as in https://iopscience.iop.org/article/10.1088/2399-6528/aab00f/pdf
/// https://github.com/alisw/AliPhysics/blob/523f2dc8b45d913e9b7fda9b27e746819cbe5b09/PWGPP/AliAnalysisTaskFilteredTree.h#L145
```

The data in streamers could be down-sampled using the Tsallis parametrization (valid for primary particles), as described in the article above.

* The mass of the pion is assumed for standard use
* The sampling is pt-dependent - it results in either flat 1/pt or flat pt
* The parameter fractio1Pt or fractionPt, indicates which fraction of the tracks are saved (normalized to pt 1Gev)
* Weights and fraction are saved in the streamer output, the original spectra of the particles can be restored

**Comment by miranov25** [comment]
## Example of restoring original spectra from the sample data set:
The wights and fraction will be reused in following data analysis detector, reconstruction,calibration)
```
treeTimeSeries_0->Draw("qpt>>his(100,-10,10)","10*(hasITSTPC&&writeData==1)*(weight*(pt*pt))","")
treeTimeSeries_0->Draw("qpt>>hisOrig(100,-10,10)","(hasITSTPC)","same")
```
The code below shows sampling with data reduction 10 (at 1 GeV). The spectra could be recovered within the statistical error

Exactly the same approach is used later in the AO2D for additional TPC track parameters, as discussed in WP4 and later approved in the PB.

image

**Comment by miranov25** [comment]
## New variables to be added to the time series
* ```<vx>,<vy>,<vz>``` - vertex position - using median and std  estimator -only for collision
* DCAr,DCAz for the combined tracks exactly as it si done for the DCAr, dcaz for the tpc tracks - only for combined tracks

**Comment by miranov25** [comment]
Hello @matthias-kleiner, @chiarazampolli   and @shahor02 

Your latest version of the code and the results seem fine. The remaining deviation from the tester expectations seems to be due to imperfections in the data, the alignment and the reconstruction itself.
I propose to merge them now. 

We need that code as soon as possible in production.

Marian

**Comment by shahor02** [comment]
It will be merged once passes CIs

**Comment by miranov25** [comment]
Hello @shahor02 and @chiarazampolli 

I believe the pull request is awaiting review; tests have already been completed. Can someone proceed with the review?
Regarding the bug in sqrtS, which remains an unresolved conversation, we can leave it as is for now.

Regards
Marian.

**Comment by matthias-kleiner** [comment]
> Hello @shahor02 and @chiarazampolli
> 
> I believe the pull request is awaiting review; tests have already been completed. Can someone proceed with the review? Regarding the bug in sqrtS, which remains an unresolved conversation, we can leave it as is for now.
> 
> Regards Marian.

I made the sqrts as a configurable input parameter for now

File: Detectors/TPC/workflow/src/TPCTimeSeriesSpec.cxx (hunk 1 above)
Comment (shahor02) [reply Detectors/TPC/workflow/src/TPCTimeSeriesSpec.cxx]: @matthias-kleiner sorry, just noticed your question. LHC DIP provides the energy in units of 0.12 GeV. I've added methods to get it in GeV, also the `getSqrtS()` method will return GeVs now: https://github.com/AliceO2Group/AliceO2/pull/12120.

**Comment by shahor02** [comment]
@miranov25 the fullCI should converge soon, then will merge.

File: Detectors/TPC/workflow/src/TPCTimeSeriesSpec.cxx (hunk 1 above)
Comment (matthias-kleiner) [reply Detectors/TPC/workflow/src/TPCTimeSeriesSpec.cxx]: Thank you for the clarification! I will then add it later

**Review by shahor02 — APPROVED** [review]

**Pull request merged**